### PR TITLE
core: Add support for DD_ENV

### PR DESCRIPTION
**Pull request by jdgumz**
Adds support for setting the `env` in spans from a `DD_ENV` environment variable.

This branch is based off of the existing one for https://github.com/DataDog/dd-trace-py/pull/1222, so that PR must be merged before this one.

### CODE REVIEW

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -381,9 +381,20 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
             if self._runtime_worker and span.span_type in _INTERNAL_APPLICATION_SPAN_TYPES:
                 span.set_tag('language', 'python')
 
-        # add common tags
+        # Apply default global tags
         if self.tags:
             span.set_tags(self.tags)
+
+        # Add default env and service tags
+        # DEV: These override the default tags, `DD_VERSION` takes precedence over `DD_TAGS=version:v`
```
Comment (brettlangdon) [inline ddtrace/tracer.py]:
```suggestion
        # DEV: These override the default global tags, `DD_VERSION` takes precedence over `DD_TAGS=version:v`
```

File: tests/test_global_config.py [hunk 2]
```diff
@@ -253,3 +253,21 @@ def test_dd_version(self):
...
+
+            # manual override still possible
+            c.env = "prod-staging"
+            assert c.env == "prod-staging"
+
+        # between DD_ENV and DATADOG_ENV, the former takes priority
+        with override_env(dict(DATADOG_ENV="prod")):
+            with override_env(dict(DD_ENV="prod-staging")):
+                c = Config()
+                assert c.env == "prod-staging"
```
Comment (brettlangdon) [inline tests/test_global_config.py]:
```suggestion
        with override_env(dict(DATADOG_ENV="prod", DD_ENV="prod-staging")):
            c = Config()
            assert c.env == "prod-staging"
```

File: tests/test_span.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from ddtrace.context import Context
 from ddtrace.constants import (
     ANALYTICS_SAMPLE_RATE_KEY, VERSION_KEY,
-    SERVICE_VERSION_KEY, SPAN_MEASURED_KEY
+    SERVICE_VERSION_KEY, SPAN_MEASURED_KEY, ENV_KEY
```
Comment (brettlangdon) [inline tests/test_span.py]:
```suggestion
    SERVICE_VERSION_KEY, SPAN_MEASURED_KEY, ENV_KEY,
```

**Comment by halfdan** [comment]
Did this one get missed? https://github.com/DataDog/dd-trace-py/blob/e41eca7433c2958addf62faacf757c6be4ffa0aa/ddtrace/bootstrap/sitecustomize.py#L99-L100

**Comment by Kyle-Verhoog** [comment]
Hi @halfdan,

It wasn't added here since we changed where the value was loaded to be consistent with the other environment variables.

Here's where it's defined: https://github.com/DataDog/dd-trace-py/blob/e41eca7433c2958addf62faacf757c6be4ffa0aa/ddtrace/settings/config.py#L58


Here's where it gets set on a span:

https://github.com/DataDog/dd-trace-py/blob/e41eca7433c2958addf62faacf757c6be4ffa0aa/ddtrace/tracer.py#L432-L433

Is it not working for you?

**Pull request merged**